File: src/main/java/org/mitre/pickledcanary/patterngenerator/output/steps/LookupResults.java

// Copyright (C) 2025 The MITRE Corporation All Rights Reserved

package org.mitre.pickledcanary.patterngenerator.output.steps;

import java.util.List;

public class LookupResults {

	private final int size;
	private final List<ConcreteOperand> operands;

	public LookupResults(int size, List<ConcreteOperand> operands) {
		this.size = size;
		this.operands = operands;
	}

	public int getMatchSize() {
		return this.size;
	}

	public List<ConcreteOperand> getOperands() {
		return this.operands;
	}

	@Override
	public String toString() {
		return "LookupResults(size:" + this.size + ", operands:" + this.operands.toString() + ")";
	}

}

File: search/treesearchlib/src/bitstructs/lookup_results.rs
// Copyright (C) 2023 The MITRE Corporation All Rights Reserved

use alloc::vec::Vec;
use core::fmt;

use super::ConcreteOperand;

#[derive(Debug, Clone)]
pub struct LookupResults {
    pub size: usize,
    pub operands: Vec<ConcreteOperand>,
}

#[derive(Debug, Clone)]
pub enum LookupError {
    OperandsDontMatch,
    FailedToGetSliceSize,
    NotEnoughBytes,
    InstructionBitsNotFound,
}

impl fmt::Display for LookupError {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        use LookupError::*;
        match self {
            OperandsDontMatch => write!(f, "Operand doesn't match existing value"),
            FailedToGetSliceSize => write!(f, "Failed to get slice size from bitvec"),
            NotEnoughBytes => write!(f, "Not enough bytes to possibly match"),
            InstructionBitsNotFound => {
                write!(f, "MaskAndChoose had no choice matching current bits")
            }
        }
    }
}
